test/channel-assignment-spec.js

```diff
@@ -0,0 +1,60 @@
+var testHelper = require('../etc/test-helper.js');
+
+describe('Channel assignment:', function() {
```

Comment:
The test cases in this spec were largely lifted (with modifications) from general-spec.js.